docs/examples/arrow_example.ipynb

```diff
@@ -195,42 +265,52 @@
    "source": [
     "### Convert the Arrow data to power-grid-model input data\n",
     "\n",
-    "No direct conversion from Arrow Tables to NumPy exists and a copy is always required.\n",
+    "The Arrow record batch or tables can then be converted to row based data or columnar data.\n",
+    "Converting Arrow data to columnar NumPy arrays is recommended to leverage the columnar nature of Arrow data. \n",
+    "This conversion can be done with zero-copy operations.\n",
+    "\n",
+    "Similar approach be adopted by the user to convert to row based data.\n",
     "\n",
-    "To ensure support for optional arguments and to prevent version lock, it is recommended to create an empty power-grid-model data set using `initialize_array` and then fill it with the Arrow data."
+    "```{note}\n",
+    "The option of `zero_copy_only` in the function below is added in this demo to verify no copies are made. Its usage is not mandatory to do zero copy conversion.\n",
+    "```"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 4,
+   "execution_count": 20,
    "metadata": {},
    "outputs": [
     {
      "data": {
       "text/plain": [
-       "array([(1, 10500.), (2, 10500.), (3, 10500.)],\n",
-       "      dtype={'names': ['id', 'u_rated'], 'formats': ['<i4', '<f8'], 'offsets': [0, 8], 'itemsize': 16, 'aligned': True})"
+       "{'id': array([1, 2, 3]), 'u_rated': array([10500., 10500., 10500.])}"
       ]
      },
-     "execution_count": 4,
+     "execution_count": 20,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
-    "def arrow_to_numpy(data: pa.lib.Table, data_type: str, component: str) -> np.ndarray:\n",
+    "def arrow_to_numpy(\n",
+    "    data: pa.RecordBatch, dataset_type: DatasetType, component_type: ComponentType, zero_copy_only: bool = False\n",
+    ") -> np.ndarray:\n",
     "    \"\"\"Convert Arrow data to NumPy data.\"\"\"\n",
-    "    result = initialize_array(data_type, component, len(data))\n",
+    "    result = {}\n",
+    "    result_dtype = power_grid_meta_data[dataset_type][component_type].dtype\n",
     "    for name, column in zip(data.column_names, data.columns):\n",
-    "        if name in result.dtype.names:\n",
-    "            result[name] = column.to_numpy()\n",
+    "        column_data = column.to_numpy(zero_copy_only=zero_copy_only)\n",
+    "        if zero_copy_only and column_data.dtype != result_dtype[name]:\n",
+    "            raise ValueError(ZERO_COPY_ERROR_MSG)\n",
```

Comment:
or just
```suggestion
    "        assert column_data.dtype == result_dtype[name]\n",
```
then you don't need the note or the argument. it's just for illustration purposes anyway